rts/motoko-rts/src/idl.rs

```diff
@@ -852,25 +851,21 @@ pub(crate) unsafe fn memory_compatible(
                 if n1 == 0 {
                     return false;
                 };
-                let mut tag1: u32;
-                let mut t11: i32;
-                loop {
-                    tag1 = leb128_decode(&mut tb1);
-                    t11 = sleb128_decode(&mut tb1);
-                    n1 -= 1;
-                    if variance == TypeVariance::Invariance || !(tag1 < tag2 && n1 > 0) {
-                        break;
-                    }
+                let tag1 = leb128_decode(&mut tb1);
+                let t11 = sleb128_decode(&mut tb1);
+                n1 -= 1;
+                if tag1 < tag2 {
+                    return false; // discarded field, return false
```

Comment:
```suggestion
                    return false; // discarded field
```
you are literally saying it in front of this
(several other occurrences below too)